CMakeLists.txt

```diff
@@ -1,102 +1,31 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 2.9)
 
 project(FreeGish C)
 
-set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake-extras/;${CMAKE_MODULE_PATH}")
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake-extras)
 
 option(INSTALL_FHS "Install into Linux FHS" OFF)
+set(OpenGL_GL_PREFERENCE LEGACY)
 
-#don't pollute source dir
-if(NOT ${INSTALL_FHS})
-    set(EXECUTABLE_OUTPUT_PATH "${PROJECT_SOURCE_DIR}")
-endif(NOT ${INSTALL_FHS})
-
-# Uncomment this to hardcode the data path. On the command line, you
-# would actually give -DDATAPATH='"/usr/share/games/gish"' or similar
-# but CMake does the extra quoting for us.
-#add_definitions(-DDATAPATH="/usr/share/games/gish")
-
-if(MSVC)
-    find_package(Ogg CONFIG REQUIRED)
-    find_package(Vorbis CONFIG REQUIRED)
-    find_package(libpng CONFIG REQUIRED)
-    find_package(SDL2 CONFIG REQUIRED)
-    find_package(OpenAL CONFIG REQUIRED)
-    find_package(OpenGL REQUIRED)
+find_package(SDL2 REQUIRED)
+find_package(OpenAL REQUIRED)
+find_package(OpenGL REQUIRED)
+find_package(Vorbis REQUIRED)
+find_package(PNG REQUIRED)
+find_library(LIBM m)
 
-else()
-    find_package(SDL2 REQUIRED sdl2)
-    find_package(OpenAL REQUIRED)
-    find_package(OpenGL REQUIRED)
-    find_package(Vorbis REQUIRED)
-    find_package(PNG REQUIRED)
-    find_library(M_LIBRARIES m)
+if(NOT INSTALL_FHS)
+    set(EXECUTABLE_OUTPUT_PATH "${PROJECT_SOURCE_DIR}")
 endif()
 
-
-
-include_directories(
-    ${SDL2_INCLUDE_DIRS}
-    ${OPENAL_INCLUDE_DIR}
-    ${OPENGL_INCLUDE_DIR}
-    ${Vorbis_INCLUDE_DIR}
-    ${PNG_INCLUDE_DIR}
-)
-
-if(NOT BIN_DIR)
-    if(UNIX)
-	SET(BIN_DIR "${CMAKE_INSTALL_PREFIX}/bin")
-    endif(UNIX)
-endif(NOT BIN_DIR)
-
-if(NOT DATA_DIR)
-    if(UNIX)
-	SET(DATA_DIR "${CMAKE_INSTALL_PREFIX}/share/freegish")
-    endif(UNIX)
-endif(NOT DATA_DIR)
-
-if(${INSTALL_FHS})
-    ADD_DEFINITIONS(-DDATAPATH="${DATA_DIR}")
-endif(${INSTALL_FHS})
-
+install(DIRECTORY animation DESTINATION ${CMAKE_INSTALL_DATADIR}/freegish)
```

Comment:
CMAKE_INSTALL_DATADIR is not set for me on Debian 11.

Reply:
It doesn't build at all or doesn't install?

It should be set when `CMAKE_INSTALL_PREFIX` is set, the readme instructs to set this.

https://github.com/freegish/freegish/pull/13/files#diff-752180eb8dc62c50aabe89619582363f65cb62ee49284c846d82ed7fe156cd1bR11-R27

Reply:
So I'm just running make install without root permission, so it just errors out, but it is trying to write to the wrong location:

```
wouter@wouter-gridscale:~/Personal/freegish/build$ make  install
Consolidate compiler generated dependencies of target freegish
[100%] Built target freegish
Install the project...
-- Install configuration: ""
-- Installing: /freegish/animation
CMake Error at cmake_install.cmake:54 (file):
  file INSTALL cannot make directory "/freegish/animation": No such file or
  directory.


make: *** [Makefile:100: install] Error 1
```

This was a quick test on Ubuntu 22.04.

Reply:
I'm guessing `CMAKE_INSTALL_PREFIX` was not set when running cmake?
The readme now says that for installation you should run:

```
cmake -DCMAKE_INSTALL_PREFIX=/usr/local -DINSTALL_FHS=ON
make
make install
```